### PR TITLE
Add `not_applicable_if` column to features

### LINKED ISSUE #208: Introduce `not_applicable_if` column in `features.csv` for clearer `not_applicable` rules, remove YAML file
At our most recent meeting in April 2024, we decided to introduce simple features with "yes/no" values. Some existing features will depend on these new features. 

For example, for current features `D-2` and `D-3` (_Restrictions on the phonemic structure of segment morphemes in the initial position_ and _in the final position_, respectively), a new feature will be created. It will be called something like `"Restrictions on the phonemic structure of segment morphemes"` and will only have two values: `"present"` and `"absent"`. Then both `D-2` and `D-3` will get `not_applicable` value type if `"absent"` is selected in this new feature.

## Current approach

Currently, we have [a YAML file](https://github.com/Jazyki-Mira/langworld_db_data/blob/2731f62b19e4baf97bfb101cd7eed7cd449a0e8c/data/inventories/features_not_applicable_rules.yaml) listing the rules for `not_applicable` value type in one feature depending on a specific value (called `trigger` in this file) in another feature. Note that this file follows a _top-down_ approach of stating dependencies.

> [!NOTE] 
> I'm linking to YAML file in specific commit (not on main tree) because it will be removed when this issue is closed and the link would stop working otherwise.

## Proposed approach

**I suggest creating a column `not_applicable_if` in `features.csv`**. In this column, we will put the **ID of a feature value** that **triggers the given feature to be `not_applicable`**. 

This means that the direction of dependency statement will be inverted compared to the existing YAML file, since this approach is _bottom-up_.

> [!NOTE]
> Initially I suggested creating a column `requires` in `features.csv`. In this column, I suggested putting the ID of a feature value that is required for the given feature to be applicable. 
>
> However, this doesn't work with features like `A-9` (diphthongs and tripthongs) where several values make `A-10` applicable but only one makes `A-10` `not_applicable`.
> 
> Theoretically, several values in one feature can make another feature `not_applicable`. But then we can put several value IDs in `not_applicable_if` column of the respective feature. On the other hand, putting several value IDs in `requires` column would look confusing as the default interpretation would be "requires **all** of these values to be applicable"

The name of the column suggests that I don't want to explicitly state that some of our features have "parents". It is not really important whether our features are seen as a hierarchical tree or as a flat structure with dependencies. With this column I want to introduce a minimal concept: this feature is `not_applicable` if a certain value (or maybe one of list of values) is present in another feature. No more than that.

Stating dependencies in `features.csv` will make data clearer for the end-user: for every feature, they will see whether its applicability depends on a value in a different feature. The dependency will be easier to visualize in GUI apps as well.

### PR DESCRIPTION
**Pull request by lemontree210**
closes #208

### CODE REVIEW

**Comment by MariAnShum** [comment]
In line B-5, I noticed two values in the column "not-applicable-if": "B-1-1, B-4-2". Perhaps "|" could be adopted as a delimiter there instead of comma?

File: data/inventories/features.csv [hunk 1]
```diff
@@ -1,127 +1,127 @@
...
+A-18,Additional articulatory oppositions of obstruents,Дополнительные артикуляционные противопоставления шумных согласных,,,,,
+A-19,Inventory of sonorants by manner of articulation,Инвентарь сонорных согласных по способу артикуляции,,,,,
+A-20,Inventory of sonorants by place of articulation,Инвентарь сонорных согласных по месту образования,,,,,
+A-21,Additional articulatory oppositions of sonorants,Дополнительные противопоставления сонорных согласных,,,,,
+A-220,Inventory of fricatives by place of articulation,Инвентарь фрикативных согласных по месту образования,,,,,
+B-1,Stress character,Тип ударения,,,,,
+B-2,Stress type,Вид ударения,,,,,
+B-3,Stress carrier,Носитель ударения,,,,B-1-1,
+B-4,Stress fixedness,Фиксированность ударения,,,,B-1-1,
+B-5,Fixed stress carrier,Носитель фиксированного ударения в слове,,,,"B-1-1, B-4-2",
```
Comment (MariAnShum) [inline data/inventories/features.csv:29]:
```suggestion
B-5,Fixed stress carrier,Носитель фиксированного ударения в слове,,,,B-1-1|B-4-2,

```

**Review by MariAnShum — COMMENTED** [review]
Wrote a small idea on notation in column "not-applicable-if"

File: data/inventories/features.csv (hunk 1 above)
Comment (lemontree210) [reply data/inventories/features.csv:29]: I think it's better to follow the pattern we already have in e.g. [list of doculects](https://raw.githubusercontent.com/Jazyki-Mira/langworld_db_data/feature/add-column-to-features-208/data/inventories/doculects.csv)

**Pull request merged**